test/e2e/storage/vsphere/vsphere_volume_vpxd_restart.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+/*
+	Test to verify that a volume remains attached through vpxd restart.
+
+	For the number of schedulable nodes:
+	1. Create a Volume with default options.
+	2. Create a Pod with the created Volume.
+	3. Verify that the Volume is attached.
+	4. Create a file with random contents under the Volume's mount point on the Pod.
+	5. Stop the vpxd service on the vCenter host.
+	6. Verify that the file is accessible on the Pod and that it's contents match.
+	7. Start the vpxd service on the vCenter host.
+	8. Verify that the Volume remains attached, the file is accessible on the Pod, and that it's contents match.
+	9. Delete the Pod and wait for the Volume to be detached.
+	10. Delete the Volume.
+*/
+var _ = utils.SIGDescribe("Verify Volume Attach Through vpxd Restart [Feature:vsphere][Serial][Disruptive]", func() {
+	f := framework.NewDefaultFramework("restart-vpxd")
+
+	type node struct {
+		name     string
+		kvLabels map[string]string
+		nodeInfo *NodeInfo
+	}
+
+	const (
+		labelKey        = "vsphere_e2e_label"
```

Comment:
It is better to keep unique key label with test identification. when multiple tests executed in random orders, they may overwrite node labels and thus results into unpredictable outcome. Refer the fix - https://github.com/kubernetes/kubernetes/pull/59211

you can set

```
labelKey = "vsphere_e2e_label_vpxd_restart"
```

Reply:
Addressed in 734e471.